rewindable-index/src/RewindableIndex/Storable.hs

```diff
@@ -62,12 +62,14 @@ import GHC.Generics (Generic)
 -}
 data family StorableEvent h
 
-data family StorablePoint h
+type family StorablePoint h
 
 data family StorableQuery h
 
 data family StorableResult h
 
+type family StorableMonad h :: * -> *
```

Comment:
Why `StorableMonad`, can the monad be stored?

If this is a type family tied to the type classes below, why not make it an associated type family? Or, will it ever be used separately?

Reply:
It is the monad in which the storable operations run (in this case IO). It is also one monad that is used across all 4 type classes so I would rather define it once.